### PR TITLE
Test: Support running raw query in Executor UT.

### PR DESCRIPTION
**Pull request by ywqzzy**
### What problem does this PR solve?

Issue Number: ref #4609 

Problem Summary:

### What is changed and how it works?
1. Add interface for Executor tests to run raw query.
2. Add table_infos map to store TableInfo for `compileQuery()`.
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by ywqzzy** [comment]
After this pr, we can move some of the executor tests in delta-merge-test into gtest.

**Review by SeaRise — APPROVED** [review]

File: dbms/src/Debug/MockStorage.cpp [hunk 1]
```diff
@@ -150,4 +151,29 @@ ColumnsWithTypeAndName MockStorage::getColumnsForMPPTableScan(Int64 table_id, In
...
+{
+    TableInfo table_info;
+    table_info.name = name;
+    table_info.id = getTableId(name);
+    int i = 0;
+    for (const auto & column : columns)
+    {
+        TiDB::ColumnInfo ret;
+        ret.tp = column.second;
+        ret.name = column.first;
```
Comment (Lloyd-Pottiger) [inline dbms/src/Debug/MockStorage.cpp:165]:
```suggestion
        std::tie(ret.name, ret.tp) = column;
```

**Review by Lloyd-Pottiger — APPROVED** [review]

**Pull request merged**